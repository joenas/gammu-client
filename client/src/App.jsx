import React, { Component } from 'react'
import { connect } from 'react-redux'
import { Route, Switch} from 'react-router-dom'
import ReactInterval from 'react-interval'
import { ConnectedRouter } from 'react-router-redux'

import { checkForNewItems } from './actions'

import ConversationsList from './containers/ConversationsList'
import NewConversation from './containers/NewConversation'
import Conversation from './containers/Conversation'

//const CHECK_NEW_INTERVAL = 5000
const CHECK_NEW_INTERVAL = 60000

class App extends Component {

  componentDidMount() {
    this.checkForNewItems()
  }

  checkForNewItems() {
   const { dispatch } = this.props
   dispatch(checkForNewItems())
  }

  render() {
    return (
      <ConnectedRouter history={this.props.history}>
        <div>
          <ReactInterval
            timeout={CHECK_NEW_INTERVAL}
            enabled={true}
            callback={() => this.checkForNewItems()}
          />
          <Switch>
            <Route exact path="/" component={ConversationsList}/>
            <Route path="/new" component={NewConversation}/>
            <Route path="/:id" component={Conversation}/>
          </Switch>
        </div>
      </ConnectedRouter>
    )
  }
}

export default connect()(App)
