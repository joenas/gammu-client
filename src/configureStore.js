import { createStore, applyMiddleware } from 'redux'
import thunkMiddleware from 'redux-thunk'
import { createLogger } from 'redux-logger'

import { routerMiddleware } from 'react-router-redux'

import smsApp from './reducers'

const loggerMiddleware = createLogger()

export default function configureStore(preloadedState, history) {
  const routingMiddleware = routerMiddleware(history)
  return createStore(
    smsApp,
    preloadedState,
    applyMiddleware(
      routingMiddleware,
      thunkMiddleware,
      loggerMiddleware
    )
  )
}
