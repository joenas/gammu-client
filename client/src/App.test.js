import { enableFetchMocks } from "jest-fetch-mock";
enableFetchMocks();

import React from "react";
import { render } from "react-dom";
import { create } from "react-test-renderer";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { createMemoryHistory } from "history";

import App from "./App";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const initialState = {
  messages: {
    isFetching: false,
    didInvalidate: false,
    items: {},
    ids: [],
    number: "",
    numberErrors: {}
  }
};

// TODO: spec these
const fakeItems = {
  items: {
    "1234": [
      {
        id: "931f3746fc303c26a41a68fcdc1b006a",
        datetime: "2018-08-06 10:29:59",
        number: "1234",
        text: "Random text",
        incoming: true,
        sent: true,
        read: false
      }
    ]
  },
  ids: ["1234"]
};

let store;
let component;
let history;

beforeEach(() => {
  history = createMemoryHistory();
  store = mockStore(initialState);
  fetch.resetMocks();
  fetch.mockResponses([JSON.stringify(fakeItems), { status: 200 }]);
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <App history={history} />
    </Provider>,
    document.createElement("div")
  );
});

it("renders a menu and container", () => {
  component = create(
    <Provider store={store}>
      <App history={history} />
    </Provider>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
