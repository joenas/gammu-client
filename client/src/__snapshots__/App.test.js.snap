// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a menu and container 1`] = `
<div>
  <div>
    <div
      className="ui top fixed menu borderless"
      style={
        Object {
          "boxShadow": "none",
        }
      }
    >
      <div
        className="ui container"
      >
        <h3
          className="header item"
          onClick={[Function]}
        >
          Messages
        </h3>
        <div
          className="right menu"
        >
          <a
            className="item"
            href="/new"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="blue edit large icon"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="ui container grid"
    >
      <div
        className="column"
      />
    </div>
  </div>
</div>
`;
